File: Ruby/fibonacci.py
def main():
	oper = Oper()
	print(oper.fibonacci(9))
	print(oper.operations)
	oper.operations = 0
	print(oper.betterFibo(9))
	print(oper.operations)

class Oper:
	memo = []
	operations = 0
	def __init__(self, x):
		self.operations = x
	def fibonacci(x):
		operations += 1
		if(x <= 2):
			if(x == 0): return 0
			else: return 1 
		else:
			return fibonacci(x-1) + fibonacci(x-2)
	def betterFibo(x):
		operations += 1
		if(memo[x] != None):
			return memo[x]
		else:
			if(x <= 2):
				if(x == 0):
					return 0
				else:
					return 1
			else:
				memo[x] = betterFibo(x-1) + betterFibo(x-2)
				return memo[x]
main()

File: Ruby/Fibonacci.java
public class Fibonacci
{
	public static int operations = 0;
	public static int[] memo = new int[100];
	public static void main(String[] args)
	{
		System.out.println(fibonacci(9));
		System.out.println(operations);
		operations = 0;
		System.out.println(betterFibo(9));
		System.out.println(operations);
	}
	public static int fibonacci(int x)
	{
		operations++;
		if(x <= 2)
		{
			if(x == 0)
				return 0;
			else
				return 1;
		}else{
			return fibonacci(x-1) + fibonacci(x-2);
		}
	}
	public static int betterFibo(int x)
	{
		operations++;
		if(memo[x] == 0)
		{
			if(x == 0)
			{
				memo[x] = 0;
			}else if(x <= 2)
			{
				memo[x] = 1;
			}else{
				memo[x] = betterFibo(x-1) + betterFibo(x-2);
			}
		}
		return memo[x];
	}
}